// @whcg/whcg-box-container/dist/index.es6.js
import { html, PolymerElement } from '@polymer/polymer/polymer-element.js';

class WhcgBoxContainer extends PolymerElement {

  static get template() {
    return html`
    <style>

      .gridcontainer {
        display: grid;
        grid-template-columns: 30px 1fr;
        grid-template-rows: auto;
        /*
        grid-template-areas: 
        "margin headline";
        "margin main"; 
        */
        
      }


      .box {
        background-color: #f7f7f9;
        border-radius: 5px;
        box-shadow: -2px 5px 8px #e6e8ed;
        padding: 30px;
        padding-top: 15px;
        margin: 30px;
        
      }

      .headline {
        grid-column: 2 / 3;
        grid-row: 1 / 2;
      }
      .flexbox {
        display: flex;
        grid-column: 2 / 3;
        grid-row: 2 / 3;
      }
      
      .row {  
          flex-direction: row;
      }

      .column {  
        flex-direction: column;
    }

    ::slotted(*) {
      padding-right:20px;
    }

    </style>


    <div class="gridcontainer box">
      <h2 class="headline">{{name}}</h2>
      <div id="flexbox" class="flexbox">
        <slot id="slotid"></slot>
      </div>
    </div>
  `;
  }

  static get properties() {
    return {
      column: {
        type: Boolean,
        notify: true,
        readOnly: false,
        observer: '_setDirection',
      },
      name: {
        type: String,
        notify: true,
        readOnly: false,
      },
    };
  }

  _setDirection() {
    this.$.flexbox.classList.remove('row');
    this.$.flexbox.classList.add('column');
  }
}

window.customElements.define('whcg-box-container', WhcgBoxContainer);

export { WhcgBoxContainer };
